server/filestore.go

```diff
@@ -646,6 +647,11 @@ const indexHdrSize = 7*binary.MaxVarintLen64 + hdrLen + checksumSize
 func (fs *fileStore) recoverMsgBlock(fi os.FileInfo, index uint32) (*msgBlock, error) {
 	mb := &msgBlock{fs: fs, index: index, cexp: fs.fcfg.CacheExpire}
 
+	// Check if we should be monitoring subjects.
+	if len(fs.cfg.Subjects) == 0 {
```

Comment:
Humm... I know that fs.cfg.Subjects can change (in fileStore.UpdateConfig()), so does that mean that we would then not track subject at all (because of setting noSubj=true)? Or maybe we need to update this value in all blocks in UpdateConfig(), but then it means that noSubj is no longer immutable and should be checked under lock in readPerSubjectInfo() (since that function can be invoked without the mb lock)

Reply:
I think this ok, upper layers have 1 or more subjects vs raft layer always none.